pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -0,0 +1,81 @@
+import abc
+from typing import Any
+
+import numpy as np
+
+from ._transform_graph_base import TransformGraphBase
+from ..transformations import (check_transform, invert_transform, concat)
+
+
+class TimeTransform(abc.ABC):
+    """Base class passed to TemporalTransformManager.add_transform() to deal with time.
+    """
+    
+    @abc.abstractmethod
+    def as_matrix(self, time) -> np.ndarray:
+        ...
+
+    @abc.abstractmethod
+    def check_transforms(self) -> "TimeTransform":
+        """Checks all transformation in a sequence. If checks are successful,
+        returns the original object"""
+        ...
+
+
+class StaticTransform(TimeTransform):
+    """Transformation, which does not change over time."""
+
+    def __init__(self, A2B):
+        """Initialize a StaticTransform with a transformation matrix.
+
+        Args:
+            A2B (matrix): 4x4 transformation 
+        """
+        self._A2B = A2B
+
+    def as_matrix(self, time):
+        return self._A2B
+    
+    def check_transforms(self):
+        self._A2B = check_transform(self._A2B)
+        return self
+
+
+class TemporalTransformManager(TransformGraphBase):
+
+    def __init__(self, strict_check=True, check=True):
+        self._transforms = {}
+        self._current_time = 0.0
+        super(TemporalTransformManager, self).__init__(strict_check, check)
+    
+    @property
+    def current_time(self):
+        return self._current_time
+
+    def set_time(self, time):
+        self._current_time = time
+    
+    @property
+    def transforms(self):
+        """Rigid transformations between nodes."""
+        return {tf_direction: transform.as_matrix(self.current_time) for
+                tf_direction, transform in self._transforms.items()}
```

Comment:
This is the key change, basically... 

We could even think about [cached properties](https://docs.python.org/3.8/library/functools.html#functools.cached_property) (but they unfortunately are from >=3.8). This would avoid re-computing interpolated transforms ...

Reply:
But I think it wouldn't be so easy to recompute the cached property when we change the current time, right?